ssl/ssl_x509.cc

```diff
@@ -765,6 +765,14 @@ void SSL_CTX_set_cert_store(SSL_CTX *ctx, X509_STORE *store) {
   ctx->cert_store = store;
 }
 
+void SSL_CTX_set1_cert_store(SSL_CTX *ctx, X509_STORE *store) {
+  check_ssl_ctx_x509_method(ctx);
+  if (store != nullptr) {
+    X509_STORE_up_ref(store);
+  }
+  SSL_CTX_set_cert_store(ctx, store);
```

Comment:
There should be a check in [SSL_CTX_set_cert_store](https://github.com/aws/aws-lc/blob/2352c50a6b5b95c12aed8b92c5881def3218a646/ssl/ssl_x509.cc#L764) that `ctx` is not null prior to dereferencing it.

Reply:
Actually, the check for `ctx == NULL` needs to occur prior to `X509_STORE_up_ref`, so it should be here as well.

Reply:
These functions return `void` so there's no way to inform the caller of the failure.  We should have `assert` statements, so at least for DEBUG builds consumers can be alerted to the failure.
```
assert(ctx != NULL);
```

Reply:
added assertion to `check_ssl_ctx_x509_method`